src/client.ts

```diff
@@ -662,6 +671,88 @@ export class MongoClient<U extends BaseMongoObject, L extends BaseMongoObject> {
 		await this.collectionHistoric.drop();
 	}
 
+	/**
+	 * Add a tag to an entity.
+	 * If the entity already has the tag, this method has no effect.
+	 *
+	 * @param query the query to select the entity to tag
+	 * @param userId id of the user performing the operation
+	 * @param options options to customize the tag
+	 */
+	public async addTagToOne(query: object, userId: string, options: AddTagOptions = {}): Promise<string> {
+		options.tag = options.tag || new ObjectId().toHexString();
+		const update = this.buildAddTagUpdate(userId, options);
+		await this.collection.findOneAndUpdate(
+			query,
+			update,
+			{ returnOriginal: false },
+		);
+		return options.tag;
+	}
+
+	/**
+	 * Same as addTagToOne, except the query is made by id.
+	 */
+	public async addTagToOneById(id: string, userId: string, options: AddTagOptions = {}): Promise<string> {
+		return await this.addTagToOne({ _id: MongoUtils.oid(id) }, userId, options);
+	}
+
+	/**
+	 * Same as addTagToOne, but for many entities
+	 */
+	public async addTagToMany(query: object, userId: string, options: AddTagOptions = {}): Promise<string> {
+		options.tag = options.tag || new ObjectId().toHexString();
```

Comment:
@petitcl tu peux utiliser la fonction MongoUtils.oid() plutôt ?

```suggestion
		options.tag = options.tag || MongoUtils.oid().toHexString();
```